prophet/handlers/arbitraryhttp/arbitraryhttp.go

```diff
@@ -0,0 +1,207 @@
+	if err := cbor.Unmarshal(input, &reqInput); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal input: %w", err)
+	}
+
+	// Validate the URL.
+	parsedURL, err := url.Parse(reqInput.Url)
+	if err != nil {
+		return nil, fmt.Errorf("invalid URL: %w", err)
+	}
+	if !h.isWhitelisted(parsedURL) {
+		return nil, errors.New("URL not allowed by whitelist")
+	}
+
+	// Default HTTP method to GET if not specified.
+	method := reqInput.Method
+	if method == "" {
+		method = http.MethodGet
+	}
+
+	// Create the HTTP request.
+	req, err := http.NewRequestWithContext(ctx, method, reqInput.Url, bytes.NewReader(reqInput.Body))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HTTP request: %w", err)
+	}
+	if len(reqInput.Body) > 0 {
+		req.Header.Set("Content-Type", "application/json")
+	}
+
+	// Execute the HTTP call.
+	resp, err := h.httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("HTTP request failed: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// Read the HTTP response body.
+	bodyBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	// Attempt to parse the response as JSON.
+	var jsonData interface{}
+	if err := json.Unmarshal(bodyBytes, &jsonData); err != nil {
+		// If JSON parsing fails, fall back to using the raw string.
+		jsonData = string(bodyBytes)
+	}
+
+	// Encode the JSON (or raw string) into CBOR.
+	cborBody, err := cbor.Marshal(jsonData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal response body to CBOR: %w", err)
+	}
+
+	// Prepare the output.
+	output := generated.ArbitraryHttpResponse{
+		Status: big.NewInt(int64(resp.StatusCode)),
+		Body:   cborBody,
+	}
+
+	// ABI‑encode the output.
+	encodedOutput, err := encodeOutput(output)
+	if err != nil {
+		return nil, fmt.Errorf("failed to ABI‑encode output: %w", err)
+	}
+
+	return encodedOutput, nil
+}
+
+// Verify checks that the HTTP response has a 200 status code and contains valid CBOR-encoded JSON data
+func (h *Handler) Verify(ctx context.Context, input []byte, output []byte) error {
+	decoded, err := decodeOutput(output)
+	if err != nil {
+		return fmt.Errorf("verify decode output: %w", err)
+	}
+
+	if decoded.Status.Int64() != 200 {
+		return fmt.Errorf("status code is not 200")
+	}
+
+	var body map[string]interface{}
+	err = cbor.Unmarshal(decoded.Body, &body)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal response body: %w", err)
+	}
+
+	return nil
+}
+
+// isWhitelisted checks if the URL's hostname is in the allowed whitelist.
+// If the whitelist is empty, all URLs are allowed.
+func (h *Handler) isWhitelisted(u *url.URL) bool {
+	if len(h.whitelist) == 0 {
+		return true
+	}
+	hostname := u.Hostname()
+	for _, allowed := range h.whitelist {
+		// Parse the allowed URL to get its hostname
+		allowedURL, err := url.Parse(allowed)
+		if err != nil {
+			continue
+		}
+		if hostname == allowedURL.Hostname() {
+			return true
+		}
+	}
```

Comment:
we should probably check Host instead of just Hostname()

```suggestion
	host := u.Host
	for _, allowed := range h.whitelist {
		// Parse the allowed URL to get its hostname
		allowedURL, err := url.Parse(allowed)
		if err != nil {
			continue
		}
		if host == allowedURL.Host {
			return true
		}
	}
```